docs/dev/allocation.qmd

```diff
@@ -0,0 +1,268 @@
+---
+title: "Allocation"
+---
+# Overview of allocation implementation
+In this document, the allocation workflow is explained. Below is an overview of it.
+```{mermaid}
+flowchart TD
+    subgraph update_allocation
+        direction TB
+        G[update mean flows]-->E[collect demand]
+        E-->F[allocate]
+    end
+    style update_allocation fill:#9ff
+    C(Begin)-->A[Initialization allocation]
+    A--> update_allocation --> H[\end of time?\]
+    H--> |No| update_allocation
+    H --> |Yes| D(End)
+```
+
+If allocation is used in a model, [Allocation structs](https://deltares.github.io/Ribasim/dev/allocation.html#the-allocation-struct) are created.
+The allocation struct stores the data that is needed for the calculations and the results of the calculation.
+In allocation, optimization is an essential part.
+The JuMP model is used to model and solve the optimization problems that come up in allocation.
+The [AllocationModel struct](https://deltares.github.io/Ribasim/dev/allocation.html#the-allocation-struct) is used for constructing the JuMP model.
+When an instance of `AllocationModel` is created, a JuMP optimization model is defined and initialized in the instance.
+More details on how allocation interacts with `JuMP.jl` is explained [here](https://deltares.github.io/Ribasim/dev/allocation.html#jump.jl-problem-interaction).
+
+After initialization, as the simulation starts, the allocation problem is solved and updated after every allocation timestep (which is specified in the TOML).
+In each iteration, the data needed for comuptation is updated.
```

Comment:
```suggestion
In each iteration, the data needed for computation is updated.
```
Which Iteration ? Do you mean every allocation timestep, or every priority iteration ? Be more specific.
Also 'the data needed' is very vague, certainly given the next sentence that 'the parameters'are updated. Be more specific.
Are you trying to tell the following in this sentence and the next one ?
With every allocation timestep a new optimization problem is formulated and solved, using the latest available (simulation) model conditions and forcing and demand predictions.